sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -392,7 +392,9 @@ case class InsertIntoHiveTable(
 
     // Invalidate the cache.
     sqlContext.sharedState.cacheManager.invalidateCache(table)
-    sqlContext.sessionState.catalog.refreshTable(table.catalogTable.identifier)
+    if (partition.nonEmpty) {
+      sqlContext.sessionState.catalog.refreshTable(table.catalogTable.identifier)
+    }
```

Comment:
@cloud-fan @ericl @mallman For non-partitioned parquet/orc tables, we convert them to the data source tables. Thus, it will not call `InsertIntoHiveTable`. 

I know it is a little bit confusing, but I am fine to keep it unchanged.